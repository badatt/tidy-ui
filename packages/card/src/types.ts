import { HTMLAttributes } from 'react';
import { TColor } from '@tidy-ui/types';

export interface ICardProps extends HTMLAttributes<HTMLDivElement> {
  accent?: TColor;
  rounded?: boolean;
  href?: string;
}

export interface ICardFooterProps extends HTMLAttributes<HTMLDivElement> {
  separated?: boolean;
}

export interface ICardHeaderProps extends HTMLAttributes<HTMLDivElement> {
  separated?: boolean;
  href?: string;
}
