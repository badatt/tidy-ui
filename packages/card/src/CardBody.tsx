import { styled } from '@tidy-ui/theme';

/**
 * Card body component, powered by various styles (with theming) and settings
 *
 */
const CardBody = styled.div`
  padding: 0.5rem 1rem;
  display: flex;
`;

export { CardBody };
