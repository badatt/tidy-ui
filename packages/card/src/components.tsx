import { LaunchIcon } from '@tidy-ui/commons';
import { css, styled } from '@tidy-ui/theme';
import { ICardFooterProps, ICardHeaderProps, ICardProps } from './types';

/**
 * Internal CardHeader root component
 *
 * @internal
 */
const CardHeaderRoot = styled.header<ICardHeaderProps>`
  ${({ theme: { typography, palette }, isDivided }) => css`
    position: relative;
    font-weight: ${typography.fontWeightBold};
    padding-top: 1rem;
    padding-right: 1rem;
    padding-bottom: 0.5rem;
    padding-left: 1rem;
    width: 100%;
    ${isDivided &&
    css`
      border-bottom: 1px solid ${palette.divider};
    `}
  `}
`;

/**
 * Internal styled anchor tag
 *
 * @internal
 */
const Link = styled.a`
  display: none;
  position: absolute;
  height: 2.5rem;
  width: 2.5rem;
  top: 0;
  right: 0;
`;

/**
 * Internal styled LaunchIcon
 *
 * @internal
 */
const Icon = styled(LaunchIcon)`
  ${({ theme: { palette, isDark } }) => css`
    height: 1.25rem;
    width: 1.25rem;
    top: 0.5rem;
    right: 0.5rem;
    position: absolute;
    color: ${isDark ? palette['neutral'][500] : palette['neutral'][400]};
  `}
`;

/**
 * Internal Card root component
 *
 * @internal
 */
const CardRoot = styled.section<ICardProps>`
  ${({ theme: { palette, layout }, isSharp, withAccent, withHref }) => css`
    display: flex;
    flex-direction: column;
    background-color: ${palette.background.card};
    box-shadow: ${layout.shadow};
    color: ${palette.text.primary};
    margin-bottom: 1.875rem;
    ${!isSharp &&
    css`
      border-radius: ${layout.radius};
    `}
    ${withAccent &&
    css`
      border-top: 2px solid ${palette[withAccent][600]};
    `}
    ${withHref &&
    css`
      cursor: pointer;
    `}
  `}
  &:hover ${Link} {
    display: block;
  }
`;

/**
 * Internal CardFooter root component
 *
 * @internal
 */
const CardFooterRoot = styled.footer<ICardFooterProps>`
  display: flex;
  padding-top: 0.5rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
  padding-left: 1rem;
  ${({ theme: { palette }, isDivided }) => css`
    ${isDivided &&
    css`
      border-top: 1px solid ${palette.divider};
    `}
  `}
`;

/**
 * Internal CardBody root component
 *
 * @internal
 */
const CardBodyRoot = styled.div`
  padding: 0.5rem 1rem;
  display: flex;
`;

export { CardBodyRoot, CardFooterRoot, CardHeaderRoot, CardRoot, Icon, Link };
