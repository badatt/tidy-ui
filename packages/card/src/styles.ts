import { createFontStyle, css } from '@tidy-ui/commons';
import { ICardProps } from './types';

const cardStyles = css<ICardProps>`
  ${({ theme: { palette, layout }, isSharp, accent, accentPosition, href, margin, height, width }) => css`
    ${createFontStyle()}
    background-color: ${palette.background.card};
    box-shadow: ${layout.shadow};
    color: ${palette.text.primary};
    margin: ${margin};
    height: ${height};
    width: ${width};
    ${!isSharp &&
    css`
      border-radius: ${layout.radius};
    `}
    ${accent && `border-${accentPosition}: 2px solid ${palette[accent][600]};`}
    ${href &&
    css`
      cursor: pointer;
    `}
  `}
`;

export { cardStyles };
