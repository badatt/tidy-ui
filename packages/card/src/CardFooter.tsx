import React from 'react';
import { CardFooterRoot } from './components';
import { ICardFooterProps } from './types';

/**
 * CardFooter component, powered by various styles (with theming) and settings
 *
 */
const CardFooter = React.forwardRef<HTMLDivElement, ICardFooterProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <CardFooterRoot role="presentation" ref={ref} {...rest}>
      {children}
    </CardFooterRoot>
  );
});

CardFooter.defaultProps = {
  divided: false,
};

CardFooter.displayName = 'CardFooter';

export { CardFooter };
