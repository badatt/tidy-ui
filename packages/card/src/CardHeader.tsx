import React, { forwardRef } from 'react';
import { CardHeaderRoot, Icon, Link } from './components';
import { ICardHeaderProps } from './types';

/**
 * CardHeader component, powered by various styles (with theming) and settings
 *
 */
const CardHeader = forwardRef<HTMLDivElement, ICardHeaderProps>((props, ref) => {
  const { children, withHref, ...rest } = props;
  return (
    <CardHeaderRoot ref={ref} {...rest}>
      {children}
      {withHref && (
        <Link href={withHref} target="_blank">
          <Icon {...rest} />
        </Link>
      )}
    </CardHeaderRoot>
  );
});

CardHeader.defaultProps = {
  isDivided: false,
};

export { CardHeader };
