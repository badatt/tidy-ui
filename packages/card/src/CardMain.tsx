import React from 'react';
import { CardMainRoot } from './components';
import { ICardProps } from './types';

/**
 * Card component powered by various styles (with theming) and settings. Can be used just
 * as is with a nice elevation effect that looks like a card on screen
 *
 */
const CardMain = React.forwardRef<HTMLDivElement, ICardProps>((props, ref) => {
  const { children, href, ...rest } = props;
  return (
    <CardMainRoot role="article" ref={ref} href={href} {...rest}>
      {href ? <a href={href}>{children}</a> : <>{children}</>}
    </CardMainRoot>
  );
});

CardMain.defaultProps = {
  accentPosition: 'top',
  sharp: false,
};

CardMain.displayName = 'CardMain';

export { CardMain };
