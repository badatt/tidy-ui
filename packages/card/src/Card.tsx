import React from 'react';
import { CardBody } from './CardBody';
import { CardFooter } from './CardFooter';
import { CardHeader } from './CardHeader';
import { CardRoot } from './components';
import { ICardProps } from './types';

/** @internal */
interface CardComponent extends React.ForwardRefExoticComponent<ICardProps & React.RefAttributes<HTMLDivElement>> {
  /** @internal */
  Body: typeof CardBody;
  /** @internal */
  Footer: typeof CardFooter;
  /** @internal */
  Header: typeof CardHeader;
}

/**
 * Card component powered by various styles (with theming) and settings. Can be used just
 * as is with a nice elevation effect that looks like a card on screen
 *
 */
const Card = React.forwardRef<HTMLDivElement, ICardProps>((props, ref) => {
  const { children, ele, href, ...rest } = props;
  const nodes = ele ? React.cloneElement(ele, {}, children) : children;
  return (
    <CardRoot role="article" ref={ref} href={href} {...rest}>
      {href ? <a href={href}>{nodes}</a> : <>{nodes}</>}
    </CardRoot>
  );
}) as CardComponent;

Card.defaultProps = {
  accentPosition: 'top',
  isSharp: false,
};

Card.displayName = 'Card';

Card.Body = CardBody;
Card.Header = CardHeader;
Card.Footer = CardFooter;

export { Card };
