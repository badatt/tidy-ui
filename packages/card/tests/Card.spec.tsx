/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight } from '@tidy-ui/commons';
import { Tone } from '@tidy-ui/types';
import { Card } from '../src';

afterEach(cleanup);
const originalError = console.error;

describe('Card', () => {
  afterEach(() => (console.error = originalError));
  it('Basic render', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Card>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Et totam eius ducimus unde voluptatibus asperiores
          fuga accusantium voluptates non nam sit beatae, modi quas animi autem aliquid. Exercitationem, veniam
          sapiente!
        </Card>
        <Card.Main>
          <Card.Header>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Basic render in dark mode', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Card>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Et totam eius ducimus unde voluptatibus asperiores
          fuga accusantium voluptates non nam sit beatae, modi quas animi autem aliquid. Exercitationem, veniam
          sapiente!
        </Card>
        <Card.Main>
          <Card.Header>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Card with sharp corners', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Card sharp>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni unde ad aspernatur iusto consequatur atque
          aperiam itaque, nulla ratione nobis quae maiores dicta possimus incidunt nesciunt minima accusamus cum
          aliquam?
        </Card>
        <Card.Main sharp>
          <Card.Header>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('card with link', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Card href="https://google.com">
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Et aperiam fugit ad doloremque architecto qui
          cupiditate? Exercitationem deleniti repellat sint illum veniam, maxime odit voluptatibus culpa iste.
          Accusantium, placeat illo!
        </Card>
        <Card.Main>
          <Card.Header href="https://google.com">Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
        <Card.Main href="https://google.com">
          <Card.Header>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('card with link in dark mode', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Card href="https://google.com">
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Et aperiam fugit ad doloremque architecto qui
          cupiditate? Exercitationem deleniti repellat sint illum veniam, maxime odit voluptatibus culpa iste.
          Accusantium, placeat illo!
        </Card>
        <Card.Main>
          <Card.Header href="https://google.com">Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
        <Card.Main href="https://google.com">
          <Card.Header>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Card with accent variants', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        {Object.values(Tone)
          .filter((i) => !isNaN(Number(i)))
          .map((v, i) => (
            <Card accent={Tone[v]} key={i}>
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Natus culpa odit excepturi iure accusamus? Ullam
              quia, quasi dignissimos expedita laudantium perferendis consectetur animi error adipisci, dolore ea
              voluptatem necessitatibus eligendi?
            </Card>
          ))}
        {Object.values(Tone)
          .filter((i) => !isNaN(Number(i)))
          .map((v, i) => (
            <Card.Main accent={Tone[v]} key={i} accentPosition="left">
              <Card.Header>Lorem ipsum dolor sit.</Card.Header>
              <Card.Body>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
                neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
              </Card.Body>
              <Card.Footer>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
            </Card.Main>
          ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Card divided', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Card.Main>
          <Card.Header divided>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati nam assumenda pariatur reiciendis
            deleniti libero ipsum! Eum tenetur maxime ratione obcaecati distinctio, odio mollitia provident optio
            debitis magni, fugiat consequuntur.
          </Card.Body>
          <Card.Footer divided>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
        <Card.Main sharp>
          <Card.Header divided>Lorem ipsum dolor sit.</Card.Header>
          <Card.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer divided>Lorem ipsum dolor sit amet consectetur.</Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('With wrapper element', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Card ele={<div />}>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Et totam eius ducimus unde voluptatibus asperiores
          fuga accusantium voluptates non nam sit beatae, modi quas animi autem aliquid. Exercitationem, veniam
          sapiente!
        </Card>
        <Card.Main>
          <Card.Header ele={<div />} divided>
            Lorem ipsum dolor sit.
          </Card.Header>
          <Card.Body ele={<div />}>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nobis, voluptatibus dolorum! Quas, natus! Est,
            neque deserunt consequatur voluptatum dolores quae! Error explicabo omnis odio molestias.
          </Card.Body>
          <Card.Footer ele={<div />} divided>
            Lorem ipsum dolor sit amet consectetur.
          </Card.Footer>
        </Card.Main>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  describe('Invalid text children for CardMain', () => {
    let consoleOutput: string[] = [];
    const mockedError = (output) => consoleOutput.push(output);
    beforeEach(() => (console.error = mockedError));

    it('Invalid text children for CardMain', () => {
      const tree = render(
        <ThemeProvider theme={orchidLight}>
          <Card.Main>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Fuga praesentium commodi quisquam eveniet beatae,
            excepturi aperiam pariatur dignissimos consequatur, iure tempore dolor nobis, delectus temporibus totam
            debitis saepe maiores minima.
          </Card.Main>
        </ThemeProvider>,
      );
      expect(tree).toMatchSnapshot();
      expect(consoleOutput[0]).toEqual(`Warning: Failed %s type: %s%s`);
    });
  });

  describe('Invalid children for CardMain', () => {
    let consoleOutput: string[] = [];
    const mockedError = (output) => consoleOutput.push(output);
    beforeEach(() => (console.error = mockedError));

    it('Invalid children for CardMain', () => {
      const tree = render(
        <ThemeProvider theme={orchidLight}>
          <Card.Main>
            <div>
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Fuga praesentium commodi quisquam eveniet
              beatae, excepturi aperiam pariatur dignissimos consequatur, iure tempore dolor nobis, delectus temporibus
              totam debitis saepe maiores minima.
            </div>
            <div>
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Fuga praesentium commodi quisquam eveniet
              beatae, excepturi aperiam pariatur dignissimos consequatur, iure tempore dolor nobis, delectus temporibus
              totam debitis saepe maiores minima.
            </div>
          </Card.Main>
        </ThemeProvider>,
      );
      expect(tree).toMatchSnapshot();
      expect(consoleOutput[0]).toEqual(`Warning: Failed %s type: %s%s`);
      console.error = originalError;
    });
  });
});
