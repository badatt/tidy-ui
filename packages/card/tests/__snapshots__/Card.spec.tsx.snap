// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Default render 1`] = `
<section
  className=""
>
  <header
    className=""
  >
    Lorem ipsum dolor sit amet consectetur.
  </header>
  <div
    className=""
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. A in libero accusantium sunt dolore repellat molestiae animi quod, cum dicta, quis rem fugiat harum quasi hic tempora aliquid! Aperiam, consectetur. Quisquam fuga eveniet et molestias beatae excepturi eaque porro officia! Veritatis, necessitatibus quam quibusdam officiis quia tempora quidem voluptas, nihil ipsam reprehenderit, veniam accusantium. Rerum obcaecati dolorum doloremque quod illum. Doloremque optio debitis enim nam unde quod ducimus, nesciunt obcaecati aspernatur aliquam consectetur a provident aliquid quasi officia cupiditate rerum necessitatibus distinctio blanditiis incidunt amet hic cum dolorum assumenda? Recusandae! Ipsa et, esse enim quae laudantium placeat sequi nisi est consectetur omnis. Ducimus, vitae fuga? Dolorem dolor, qui nisi laudantium tenetur totam id quisquam voluptatem maxime molestiae libero provident commodi?
  </div>
  <footer
    className=""
  >
    Lorem ipsum dolor sit amet.
  </footer>
</section>
`;

exports[`Card card with accent 1`] = `
<section
  className=""
  href="https://google.com"
>
  <a
    href="https://google.com"
  >
    <header
      className=""
    >
      Lorem ipsum dolor sit amet consectetur.
      <a
        className=""
        href="https://google.com"
        target="_blank"
      >
        <svg
          className=""
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3a1 1 0 100 2h2.586l-6.293 6.293a1 1 0 101.414 1.414L15 6.414V9a1 1 0 102 0V4a1 1 0 00-1-1h-5z"
          />
          <path
            d="M5 5a2 2 0 00-2 2v8a2 2 0 002 2h8a2 2 0 002-2v-3a1 1 0 10-2 0v3H5V7h3a1 1 0 000-2H5z"
          />
        </svg>
      </a>
    </header>
    <div
      className=""
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. A in libero accusantium sunt dolore repellat molestiae animi quod, cum dicta, quis rem fugiat harum quasi hic tempora aliquid! Aperiam, consectetur. Quisquam fuga eveniet et molestias beatae excepturi eaque porro officia! Veritatis, necessitatibus quam quibusdam officiis quia tempora quidem voluptas, nihil ipsam reprehenderit, veniam accusantium. Rerum obcaecati dolorum doloremque quod illum. Doloremque optio debitis enim nam unde quod ducimus, nesciunt obcaecati aspernatur aliquam consectetur a provident aliquid quasi officia cupiditate rerum necessitatibus distinctio blanditiis incidunt amet hic cum dolorum assumenda? Recusandae! Ipsa et, esse enim quae laudantium placeat sequi nisi est consectetur omnis. Ducimus, vitae fuga? Dolorem dolor, qui nisi laudantium tenetur totam id quisquam voluptatem maxime molestiae libero provident commodi?
    </div>
    <footer
      className=""
    >
      Lorem ipsum dolor sit amet.
    </footer>
  </a>
</section>
`;

exports[`Card card with link 1`] = `
<section
  className=""
  href="https://google.com"
>
  <a
    href="https://google.com"
  >
    <header
      className=""
    >
      Lorem ipsum dolor sit amet consectetur.
      <a
        className=""
        href="https://google.com"
        target="_blank"
      >
        <svg
          className=""
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3a1 1 0 100 2h2.586l-6.293 6.293a1 1 0 101.414 1.414L15 6.414V9a1 1 0 102 0V4a1 1 0 00-1-1h-5z"
          />
          <path
            d="M5 5a2 2 0 00-2 2v8a2 2 0 002 2h8a2 2 0 002-2v-3a1 1 0 10-2 0v3H5V7h3a1 1 0 000-2H5z"
          />
        </svg>
      </a>
    </header>
    <div
      className=""
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. A in libero accusantium sunt dolore repellat molestiae animi quod, cum dicta, quis rem fugiat harum quasi hic tempora aliquid! Aperiam, consectetur. Quisquam fuga eveniet et molestias beatae excepturi eaque porro officia! Veritatis, necessitatibus quam quibusdam officiis quia tempora quidem voluptas, nihil ipsam reprehenderit, veniam accusantium. Rerum obcaecati dolorum doloremque quod illum. Doloremque optio debitis enim nam unde quod ducimus, nesciunt obcaecati aspernatur aliquam consectetur a provident aliquid quasi officia cupiditate rerum necessitatibus distinctio blanditiis incidunt amet hic cum dolorum assumenda? Recusandae! Ipsa et, esse enim quae laudantium placeat sequi nisi est consectetur omnis. Ducimus, vitae fuga? Dolorem dolor, qui nisi laudantium tenetur totam id quisquam voluptatem maxime molestiae libero provident commodi?
    </div>
    <footer
      className=""
    >
      Lorem ipsum dolor sit amet.
    </footer>
  </a>
</section>
`;

exports[`Card dark mode card 1`] = `
<section
  className=""
  href="https://google.com"
>
  <a
    href="https://google.com"
  >
    <header
      className=""
    >
      Lorem ipsum dolor sit amet consectetur.
      <a
        className=""
        href="https://google.com"
        target="_blank"
      >
        <svg
          className=""
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3a1 1 0 100 2h2.586l-6.293 6.293a1 1 0 101.414 1.414L15 6.414V9a1 1 0 102 0V4a1 1 0 00-1-1h-5z"
          />
          <path
            d="M5 5a2 2 0 00-2 2v8a2 2 0 002 2h8a2 2 0 002-2v-3a1 1 0 10-2 0v3H5V7h3a1 1 0 000-2H5z"
          />
        </svg>
      </a>
    </header>
    <div
      className=""
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. A in libero accusantium sunt dolore repellat molestiae animi quod, cum dicta, quis rem fugiat harum quasi hic tempora aliquid! Aperiam, consectetur. Quisquam fuga eveniet et molestias beatae excepturi eaque porro officia! Veritatis, necessitatibus quam quibusdam officiis quia tempora quidem voluptas, nihil ipsam reprehenderit, veniam accusantium. Rerum obcaecati dolorum doloremque quod illum. Doloremque optio debitis enim nam unde quod ducimus, nesciunt obcaecati aspernatur aliquam consectetur a provident aliquid quasi officia cupiditate rerum necessitatibus distinctio blanditiis incidunt amet hic cum dolorum assumenda? Recusandae! Ipsa et, esse enim quae laudantium placeat sequi nisi est consectetur omnis. Ducimus, vitae fuga? Dolorem dolor, qui nisi laudantium tenetur totam id quisquam voluptatem maxime molestiae libero provident commodi?
    </div>
    <footer
      className=""
    >
      Lorem ipsum dolor sit amet.
    </footer>
  </a>
</section>
`;

exports[`Card not rounded 1`] = `
<section
  className=""
>
  <header
    className=""
  >
    Lorem ipsum dolor sit amet consectetur.
  </header>
  <div
    className=""
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. A in libero accusantium sunt dolore repellat molestiae animi quod, cum dicta, quis rem fugiat harum quasi hic tempora aliquid! Aperiam, consectetur. Quisquam fuga eveniet et molestias beatae excepturi eaque porro officia! Veritatis, necessitatibus quam quibusdam officiis quia tempora quidem voluptas, nihil ipsam reprehenderit, veniam accusantium. Rerum obcaecati dolorum doloremque quod illum. Doloremque optio debitis enim nam unde quod ducimus, nesciunt obcaecati aspernatur aliquam consectetur a provident aliquid quasi officia cupiditate rerum necessitatibus distinctio blanditiis incidunt amet hic cum dolorum assumenda? Recusandae! Ipsa et, esse enim quae laudantium placeat sequi nisi est consectetur omnis. Ducimus, vitae fuga? Dolorem dolor, qui nisi laudantium tenetur totam id quisquam voluptatem maxime molestiae libero provident commodi?
  </div>
  <footer
    className=""
  >
    Lorem ipsum dolor sit amet.
  </footer>
</section>
`;
